docs/review.md

```diff
@@ -0,0 +1,259 @@
+
+# Step 1:  How far did they get?
+
+* Features
+  * [ ] Can store tasks
+  * [ ] Instant updates
+  * [ ] Marking tasks as done
+
+* Bonus Features
+  * [ ] Filters
+  * [ ] Display number of tasks
+  * [ ] Clearing completed tasks
+  * [ ] Deploy the app
+  * [ ] Persistence layer with Node
+  * [ ] CSS Awesomeness
+
+# Step 2: Structure and supporting files
+
+# Installation/Usage Instructions in README (+)
+
+As we have seen previously, the README is a great place to show the full story of how your app is used (from a user's perspective).  Include instructions for how to download and run the tests, run the app, e.g.:
+
+```sh
+$ git clone git@github.com:[USERNAME]/todo-challenge.git
+$ cd todo-challenge
+$ ???
+$ ???
+```
+
+To run app:
+
+```sh
+$ ???
+$ open ???
+```
+
+It's a great idea to include some screenshots in your README.  For more info on embedding images in a README: https://guides.github.com/features/mastering-markdown/
+
+e.g.:
+```
+![Screenshot](https://path_to_your_image)
+```
+
+You will need to host your images somewhere, e.g.:
+* http://imgur.com/
+* http://dropbox.com/
+
+
+## If using a framework, delete any unnecessary code
+
+## Clean up
+
+* Remove comments
+* Try to avoid having log statements in your production code
+
+## ensure package.json devDependencies lists bower as a dependency (++)
+
+https://github.com/makersacademy/todo_challenge/issues/68 related?
+
+# Step 3: Tests
+
+## Tests with no expectation
+
+## Tests with multiple expectation
+
+Try not to have multiple expectations in unit tests. You might want a different test for the error message.
+
+## No integration tests, or poor test coverage
+
+# Step 4: Application code
+
+## Keep data in controller or service
+
+Rather than:
+```javascript
+// this variable is declared in a global scope
+var defaultList = { "items": [ ] }
+
+todoApp.controller('toDoCtrl', function() {
+// and then its data is mutated within a controller
+  this.addItem = function(item) {
+    defaultList.items.push(item)
+  }
+  
+});
+```
+Prefer declaring `defaultList` within the controller:
+```javascript
+todoApp.controller('toDoCtrl', function() {
+  var defaultList = { "items": [ ] }
+  
+  this.addItem = function(item) {
+    defaultList.items.push(item)
+  }
+});
+```
+
+## Prefer list array outside of function
+
+```javascript
+toDo.controller('toDoController', [function() {
+  var self = this;
+
+  self.addItem = function() {
+    if (self.list === undefined) {
+      self.list = [];
+    }
+    var item = {
+      completed: false,
+      task: self.toDoItem
+    }
+    self.list.push(item)
+  };
+```
+
+could be
+
+```javascript
+toDo.controller('toDoController', [function() {
+  var self = this;
+  if (self.list === undefined) {
```

Comment:
I think you don't need the `if` statement because it only gets run once so it would be initialize properly